### PR TITLE
Adds app:scaffold and app:run commands

### PR DESCRIPTION
**Pull request by Aschen**
## What does this PR do?

Adds the `kourou app:scaffold <app name>` and `kourou app:run` commands.

The `app:scaffold` command creates a new directory and copy the `package.json` and the `app.ts` files into it. Then it install the latest Kuzzle version and the development dependencies needed for Typescript

It will creates the following structure for the new project
image

## Other changes

 - remove `execa` dependency
 - support windows for path

### CODE REVIEW

File: src/commands/app/run.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  static description = 'Run the Kuzzle application in the current directory'
+
+  static flags = {
+    help: flags.help(),
+  }
+
+  async runSafe() {
+    await AppStartServices.run()
+
+    await execute('node', '-r', 'ts-node/register', 'app.ts')
```
Comment (xbill82) [inline src/commands/app/run.ts:22]: What's the benefit of this command versus putting an `npm run dev` script (running the same command) into the app scaffold?

File: src/commands/app/scaffold.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    help: flags.help(),
+  }
+
+  static args = [
+    { name: 'name', description: 'Application name', required: true },
+  ]
+
+  async runSafe() {
+    this.logInfo(`Scaffold a new Kuzzle application in ${this.args.name}/`)
+    await execute('rm', '-rf', this.args.name)
```
Comment (xbill82) [inline src/commands/app/scaffold.ts:27]: U sure about this? Wouldn't it be better to throw an error and let the user delete the directory?

File: src/support/execute.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { spawn } from 'child_process'
+
+export async function execute(...args: any[]): Promise<any> {
```
Comment (xbill82) [inline src/support/execute.ts:3]: Aren't we re-inventing [execa](https://www.npmjs.com/package/execa) here?

File: src/commands/app/scaffold.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+    await execute('mkdir', this.args.name)
+
+    const templatePath = path.join(templatesDir, 'app-scaffold', 'ts')
+    await this.renderTemplates(templatePath, templatePath)
+
+    this.logInfo('Installing latest Kuzzle version via NPM...')
+
+    // await execute('npm', 'install', 'kuzzle', { cwd: this.args.name })
+    // await execute('npm', 'install', { cwd: this.args.name })
```
Comment (rolljee) [inline src/commands/app/scaffold.ts:36]: Commented code should be removed

File: src/support/execute.ts (hunk 3 above)
Comment (Aschen) [reply src/support/execute.ts:3]: I took in the other way, I removed `execa` :-)

Comment (xbill82) [reply src/support/execute.ts]: Why?

Comment (Aschen) [reply src/support/execute.ts]: To reduce the number of dependencies installed with the CLI.

**Review by scottinet — APPROVED** [review]

**Review by xbill82 — APPROVED** [review]

**Pull request merged**